datafusion/common/src/pyarrow.rs

```diff
@@ -138,6 +138,9 @@ mod tests {
     fn test_py_scalar() {
         init_python();
 
+        // TODO: remove this attribute when bumping pyo3 to v0.23.0
+        // See: <https://github.com/PyO3/pyo3/blob/v0.23.0/guide/src/migration.md#gil-refs-feature-removed>
+        #[allow(unexpected_cfgs)]
```

Comment:
pyo3 will be updated in 
- https://github.com/apache/datafusion/pull/13663

I need to find some time to polish that PR a bit